cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/ShardedHBaseQueueStrategy.java

```diff
@@ -159,16 +177,18 @@ public void close() throws IOException {
     scansExecutor.shutdownNow();
   }
 
-  private byte[] getShardedKey(ConsumerGroupConfig groupConfig, int instanceId, byte[] originalRowKey) {
+  static byte[] getShardedKey(ConsumerGroupConfig groupConfig, @Nullable Integer instanceId, byte[] originalRowKey) {
     // Need to subtract the SALT_BYTES as the row key distributor will prefix the key with salted bytes
     byte[] result = new byte[PREFIX_BYTES - SaltedHBaseQueueStrategy.SALT_BYTES + originalRowKey.length];
     Bytes.putBytes(result, PREFIX_BYTES - SaltedHBaseQueueStrategy.SALT_BYTES,
                    originalRowKey, 0, originalRowKey.length);
+    Bytes.putLong(result, 0, groupConfig.getGroupId());
 
     // Default for FIFO case.
-    int shardId = groupConfig.getDequeueStrategy() == DequeueStrategy.FIFO ? -1 : instanceId;
-    Bytes.putLong(result, 0, groupConfig.getGroupId());
-    Bytes.putInt(result, Bytes.SIZEOF_LONG, shardId);
+    if (instanceId != null) {
```

Comment:
When it will be null? And what does it mean? I don't see any doc or comment explaining it.